tests/local/e2e_test.go

```diff
@@ -40,51 +39,61 @@ func TestE2E(t *testing.T) {
 
 // Define the Teleporter before and after suite functions.
 var _ = ginkgo.BeforeSuite(func() {
+	// Generate the Teleporter deployment values
+	teleporterDeployerTransaction, teleporterDeployedBytecode, teleporterDeployerAddress, teleporterContractAddress, err :=
+		deploymentUtils.ConstructKeylessTransaction(
+			teleporterByteCodeFile,
+			false,
+			deploymentUtils.GetDefaultContractCreationGasPrice(),
+		)
+	Expect(err).Should(BeNil())
+
 	// Create the local network instance
 	LocalNetworkInstance = NewLocalNetwork(
 		"teleporter-test-local-network",
 		warpGenesisTemplateFile,
 		[]SubnetSpec{
 			{
-				Name:       "A",
-				EVMChainID: 12345,
-				NodeCount:  2,
+				Name:                       "A",
+				EVMChainID:                 12345,
+				TeleporterContractAddress:  teleporterContractAddress,
+				TeleporterDeployedBytecode: teleporterDeployedBytecode,
+				TeleporterDeployerAddress:  teleporterDeployerAddress,
+				NodeCount:                  2,
 			},
 			{
-				Name:       "B",
-				EVMChainID: 54321,
-				NodeCount:  2,
+				Name:                       "B",
+				EVMChainID:                 54321,
+				TeleporterContractAddress:  teleporterContractAddress,
+				TeleporterDeployedBytecode: teleporterDeployedBytecode,
+				TeleporterDeployerAddress:  teleporterDeployerAddress,
+				NodeCount:                  2,
 			},
 		},
 		2,
 	)
+	log.Info("Started local network")
 
-	// Generate the Teleporter deployment values
-	teleporterDeployerTransaction, teleporterDeployerAddress, teleporterContractAddress, err :=
-		deploymentUtils.ConstructKeylessTransaction(
-			teleporterByteCodeFile,
-			false,
-			deploymentUtils.GetDefaultContractCreationGasPrice(),
-		)
-	Expect(err).Should(BeNil())
-
+	// Only need to deploy Teleporter on the C-Chain since it is included in the genesis of the subnet chains.
 	_, fundedKey := LocalNetworkInstance.GetFundedAccountInfo()
-	LocalNetworkInstance.DeployTeleporterContracts(
+	LocalNetworkInstance.DeployTeleporterContractToCChain(
 		teleporterDeployerTransaction,
 		teleporterDeployerAddress,
 		teleporterContractAddress,
 		fundedKey,
-		true,
 	)
+	LocalNetworkInstance.SetTeleporterContractAddress(teleporterContractAddress)
```

Comment:
I think it was a little previously actually. There was a boolean parameter in one exported function for whether or not another exported function should be called from it. Since they are both exported, I think it's cleaner to make them independent and left up to the caller.